### PR TITLE
Fix the issue description

### LINKED ISSUE #1271: SWC 107: False positive
myth -v 4 analyze --rpc infura-mainnet -b 7 -t 3 --enable-coverage-strategy -a 0x3dAe53Ee778A324BD317BD270d6227406B6bd4eD --attacker-address 0xb9708611a63461e96b21bb8dd454fedbd74c261e --creator-address 0x56d6ceF93660bc3007492416A618e7C2dceE0c7e -l --max-depth 290

Which gives:

    ==== State change after external call ====
    SWC ID: 107
    Severity: Low
    Contract: 0x3dAe53Ee778A324BD317BD270d6227406B6bd4eD
    Function name: _function_0x3a9a0ac8
    PC address: 7540
    Estimated Gas Usage: 58465 - 275144
    The contract account state is changed after an external call. 
    Consider that the called contract could re-enter the function before this state change takes place. This can lead to business logic vulnerabilities.
    --------------------
    Initial State:
    
    Account: [SOMEGUY], balance: 0x1, nonce:0, storage:{}
    Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}
    
    Transaction Sequence:
    
    Caller: [SOMEGUY], function: unknown, txdata: 0x3a9a0ac8, value: 0x0
    
    ==== State change after external call ====
    SWC ID: 107
    Severity: Low
    Contract: 0x3dAe53Ee778A324BD317BD270d6227406B6bd4eD
    Function name: fallback
    PC address: 7568
    Estimated Gas Usage: 59624 - 276352
    The contract account state is changed after an external call. 
    Consider that the called contract could re-enter the function before this state change takes place. This can lead to business logic vulnerabilities.
    --------------------
    Initial State:
    
    Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}
    Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}
    
    Transaction Sequence:
    
    Caller: [SOMEGUY], function: unknown, txdata: 0x02000103, value: 0xe8d4a51000
    
    ==== State change after external call ====
    SWC ID: 107
    Severity: Low
    Contract: 0x3dAe53Ee778A324BD317BD270d6227406B6bd4eD
    Function name: fallback
    PC address: 7615
    Estimated Gas Usage: 58683 - 275411
    The contract account state is changed after an external call. 
    Consider that the called contract could re-enter the function before this state change takes place. This can lead to business logic vulnerabilities.
    --------------------
    Initial State:
    
    Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}
    Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}
    
    Transaction Sequence:
    
    Caller: [SOMEGUY], function: unknown, txdata: 0x, value: 0xe8d4a51000
    
    ==== State change after external call ====
    SWC ID: 107
    Severity: Low
    Contract: 0x3dAe53Ee778A324BD317BD270d6227406B6bd4eD
    Function name: _function_0x3a9a0ac8
    PC address: 7643
    Estimated Gas Usage: 58465 - 275144
    The contract account state is changed after an external call. 
    Consider that the called contract could re-enter the function before this state change takes place. This can lead to business logic vulnerabilities.
    --------------------
    Initial State:
    
    Account: [SOMEGUY], balance: 0x1, nonce:0, storage:{}
    Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}
    
    Transaction Sequence:
    
    Caller: [SOMEGUY], function: unknown, txdata: 0x3a9a0ac8, value: 0x0

Whereas the call is performed to a fixed address in storage only owner can change which in turns doesn’t call anything else.

### PR DESCRIPTION
**Pull request by norhh**
related to  #1271

### CODE REVIEW

**Comment by ytrezq** [comment]
For me, there’s no point mentioning the issue at all. Performing a state change after calling a library is fully expected and has nothing to do with reentrancy. Might just worth not mentioning anything if the address is fixed.

**Comment by norhh** [comment]
Built in contracts are filtered out while detecting these issues.

**Comment by ytrezq** [comment]
@norhh so why not filter out libraries which calls nothing else with contract provided parameters too?

**Comment by b-mueller** [comment]
@ytrezq most users check source code of a single contract in isolation which means the library code isn't usually available. Reporting a low-risk (informational) issue is a reasonable catch-all. It would be possible to build logic to also consider the behaviour of the library when the code can be obtained, but that's a lot of overhead for little gain IMO. We happily accept PRs though :)

File: mythril/analysis/modules/state_change_external_calls.py [hunk 1]
```diff
@@ -66,12 +69,15 @@ def get_issue(
...
             "[EXTERNAL_CALLS] Detected state changes at addresses: {}".format(address)
         )
-        description_head = (
-            "The contract account state is changed after an external call. "
-        )
+        read_or_write = "write"
+        if global_state.get_current_instruction()["opcode"] == "SLOAD":
+            read_or_write = "read"
+        address_type = "user defined" if self.user_defined_address else "fixed"
+        description_head = "Persistent state {} after call".format(read_or_write)
```
Comment (norhh) [inline mythril/analysis/modules/state_change_external_calls.py]: @JoranHonig, @wuestholz is this the description used by harvey?

Comment (nbanmp) [reply mythril/analysis/modules/state_change_external_calls.py]: https://github.com/ConsenSys/harvey/blob/b05a643c5e4288115ff1d8842d651b9e8ccd6320/extensions/warnings.go#L397

Yup.

Comment (norhh) [reply mythril/analysis/modules/state_change_external_calls.py]: Lol, I don't have the permissions for the repo :laughing:

**Pull request merged**